cdc/cdc.cc

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <utility>
+
+#include "cdc/cdc.hh"
+#include "database.hh"
+#include "db/config.hh"
+#include "dht/murmur3_partitioner.hh"
+#include "schema.hh"
+#include "schema_builder.hh"
+#include "service/migration_manager.hh"
+#include "service/storage_proxy.hh"
+#include "service/storage_service.hh"
+
+using seastar::sstring;
+
+namespace cdc {
+
+static sstring log_name(const sstring& table_name) {
+    static constexpr const auto cdc_log_suffix = "_scylla_cdc_log";
+    return table_name + cdc_log_suffix;
+}
+
+static sstring desc_name(const sstring& table_name) {
+    static constexpr const auto cdc_desc_suffix = "_scylla_cdc_desc";
+    return table_name + cdc_desc_suffix;
+}
+
+static future<> remove_log(const sstring& ks_name, const sstring& table_name) {
+    auto& mm = service::get_local_migration_manager();
```

Comment:
why do you introduce these intermediate variables (`mm`)? They make my brain use more power.

Reply:
By the way, since we don't like globals, wouldn't it be possible to pass the migration manager and storage_service down from the caller of CDC functions?
From what I know we're in the middle of a war with globals, but this patchset introduced a lot of global variable references.

Reply:
Intermediate variables are just to make the formatting better. It helps in making long lines shorter.

Of course, we can pass them down and use globals on the call points :). I will check how much more messy the code will be if I pass down all the globals instead of using them directly.

Reply:
> Intermediate variables are just to make the formatting better. It helps in making long lines shorter.

Most of the lines wouldn't be so long anyway (yeah I know, you want to have them <= 80 characters, which is insane :P). You can always do
```
    return service::get_local_migration_manager()
            .announce_column_family_drop(ks_name, desc_name(table_name), false);
```

which is much better for readability IMO.

Reply:
> Of course, we can pass them down and use globals on the call points :)

Preferably don't use globals at all. For example `storage_service` is passed down all the way from main in a lot of places, similarly to `storage_proxy`. I don't know about `migration_manager` or `snitch_ptr` though.

Reply:
> Preferably don't use globals at all. For example `storage_service` is passed down all the way from main in a lot of places, similarly to `storage_proxy`. I don't know about `migration_manager` or `snitch_ptr` though.

Yes. That's why I said I will see what can be done. I agree with the principles.

> which is much better for readability IMO.

Since it's a matter of taste - let's agree to disagree here :).

Reply:
> Since it's a matter of taste - let's agree to disagree here :).

But I've just shown you how to achieve short lines AND not have intermediate variables. Isn't that a win-win situation? :P

Reply:
> But I've just shown you how to achieve short lines AND not have intermediate variables. Isn't that a win-win situation? :P

I still find the intermediate variable more readable. Both solutions need two lines. Solution with no Intermediate variable requires a break that is IMO way worse than a variable just above the usage. To be honest I totally don't understand the problem with the variable so I don't think this discussion brings much value.

Reply:
In that particular place (two line function) it probably doesn't make a difference. But when the function is longer, any variable introduced early in the function makes me wonder "is that variable used anywhere else?" which consumes CPU cycles in the human brain which would be better spent elsewhere.
It can sometimes be a source of bugs too, when a variable is used by accident where it shouldn't have been (that's why I really don't like the common pattern where a variable is declared as a result of a function call and then immediately moved from; it would be better to call the function right where it's needed)

Reply:
Per your other request all those will become variables passed to the functions so I really start to worry about your brain cpu :)

Reply:
Yay!